subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <json.h>
+#include <net/fota_download.h>
+#include <net/aws_jobs.h>
+#include <net/aws_fota.h>
+#include <logging/log.h>
+
+#include "aws_fota_json.h"
+
+LOG_MODULE_REGISTER(aws_fota, CONFIG_AWS_FOTA_LOG_LEVEL);
+
+/* Enum to keep the fota status */
+enum fota_status {
+	NONE = 0,
+	DOWNLOAD_FIRMWARE,
+	APPLY_FIRMWARE,
+};
+
+/* Map of fota status to report back */
+static const char * const fota_status_strings[] = {
+	[DOWNLOAD_FIRMWARE] = "download_firmware",
+	[APPLY_FIRMWARE] = "apply_update",
```

Comment:
`"apply_firmware"`

Reply:
This should actually have been `APPLY_UPDATE` my fault. They keep changing the naming in cloud.

Reply:
Tnx for spotting changed now.